### PR TITLE
Updated Q feature dev name

### PR DESCRIPTION
**Pull request by kumsmrit**
## Problem
QFeature Dev was renamed to Amazon Q Developer Agent for software development; so updated the feature name for chat messages and logs.
## Solution
Updated the Q feature dev name for chat messages and logs.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sarayev — APPROVED** [review]

**Review by sannicm — APPROVED** [review]

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 1]
```diff
@@ -726,12 +727,21 @@ export class FeatureDevController {
...
             }
         } catch (err: any) {
-            this.messenger.sendErrorMessage(
-                createUserFacingErrorMessage(err.message),
-                message.tabID,
-                this.retriesRemaining(session),
-                session?.state.phase
-            )
+            if (err instanceof WorkspaceFolderNotFoundError) {
+                this.messenger.sendAnswer({
```
Comment (justinmk3) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:731]: how is this change related?

Comment (kumsmrit) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:731]: This has been updated based on UX feedback [here](https://quip-amazon.com/fA3JAlEEK3ab/QFeatureDev-Errors-string-review#temp:C:SAH3f48f3e68c914276b0c7517c6) as the older message contained feature name which was required to be removed.

**Review by sarayev — APPROVED** [review]

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts (hunk 1 above)
Comment (sannicm) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:753]: Is the WorkspaceFolderNotFoundError actually possible here?

Comment (kumsmrit) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:753]: Yes, this is where `WorkspaceFolderNotFoundError` will be thrown for case when workspace is not open.

Comment (sannicm) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:753]: I think is a valid remark then the one from justin, it feels like it is being combined two features into one. The rename and the handling of this new error. I approve anyway

Comment (kumsmrit) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:753]: This was changed as we do not want to show feature name in case of this error message.

File: packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts [hunk 2]
```diff
@@ -440,7 +441,7 @@ export class FeatureDevController {
             if (!this.isAmazonQVisible) {
                 const open = 'Open chat'
                 const resp = await vscode.window.showInformationMessage(
-                    'Your code suggestions from Amazon Q are ready to review',
+                    'The Amazon Q Developer Agent for software development has generated code for you to review',
```
Comment (hayemaxi) [inline packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:444]: IMO: this is quite verbose

Comment (kumsmrit) [reply packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts:444]: This text is suggested by AWS documentation team [here](https://quip-amazon.com/fA3JAlEEK3ab/QFeatureDev-Errors-string-review#temp:C:SAH3f48f3e68c914276b0c7517c6)

**Review by hayemaxi — APPROVED** [review]
